src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -180,47 +146,48 @@ const FixPhoneNumbers: React.FC = () => {
     numberIndex: number,
     event: React.ChangeEvent<HTMLInputElement>,
   ): void => {
-    const temp = [...test];
-    test[contactIndex].numbers[numberIndex].number = event.target.value;
-    setTest(temp);
+    const temp = [...dataState];
+    dataState[contactIndex].phoneNumbers.nodes[numberIndex].number =
+      event.target.value;
```

Comment:
Usually, I prefer to pass just the value to the function instead of the event, but it should be fine.